python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/aaa-accounting.j2

```diff
@@ -6,48 +6,50 @@
-{%                 if command_default.group is arista.avd.defined %}
-{%                     set aaa_accounting_commands_commands_console_cli = aaa_accounting_commands_commands_console_cli ~ " group " ~ command_default.group %}
-{%                 endif %}
-{%                 if command_default.logging is arista.avd.defined(true) %}
-{%                     set aaa_accounting_commands_commands_console_cli = aaa_accounting_commands_commands_console_cli ~ " logging" %}
+{%                 if command_default.type is arista.avd.defined('none') %}
+aaa accounting commands {{ command_default.commands }} console none
+{%                 elif command_default.group is arista.avd.defined or command_default.logging is arista.avd.defined(true) %}
+{%                     set commands_console_cli = "aaa accounting commands " ~ command_default.commands ~ " console " ~ command_default.type %}
+{%                     if command_default.group is arista.avd.defined %}
+{%                         set commands_console_cli = commands_console_cli ~ " group " ~ command_default.group %}
+{%                     endif %}
+{%                     if command_default.logging is arista.avd.defined(true) %}
+{%                         set commands_console_cli = commands_console_cli ~ " logging" %}
+{%                     endif %}
+{{ commands_console_cli }}
 {%                 endif %}
 {%             endif %}
-{{ aaa_accounting_commands_commands_console_cli }}
 {%         endfor %}
 {%     endif %}
 {%     if aaa_accounting.exec.default.type is arista.avd.defined %}
-{%         set aaa_accounting_exec_default_cli = "aaa accounting exec default" %}
 {%         if aaa_accounting.exec.default.type is arista.avd.defined('none') %}
-{%             set aaa_accounting_exec_default_cli = aaa_accounting_exec_default_cli ~ " none" %}
-{%         else %}
-{%             set aaa_accounting_exec_default_cli = aaa_accounting_exec_default_cli ~ " " ~ aaa_accounting.exec.default.type %}
+aaa accounting exec default none
+{%         elif aaa_accounting.exec.default.group is arista.avd.defined or aaa_accounting.exec.default.logging is arista.avd.defined(true) %}
+{%             set exec_default_cli = "aaa accounting exec default " ~ aaa_accounting.exec.default.type %}
 {%             if aaa_accounting.exec.default.group is arista.avd.defined %}
-{%                 set aaa_accounting_exec_default_cli = aaa_accounting_exec_default_cli ~ " group " ~ aaa_accounting.exec.default.group %}
+{%                 set exec_default_cli = exec_default_cli ~ " group " ~ aaa_accounting.exec.default.group %}
 {%             endif %}
 {%             if aaa_accounting.exec.default.logging is arista.avd.defined(true) %}
-{%                 set aaa_accounting_exec_default_cli = aaa_accounting_exec_default_cli ~ " logging" %}
+{%                 set exec_default_cli = exec_default_cli ~ " logging" %}
 {%             endif %}
+{{ exec_default_cli }}
```

Comment:
I feel this could be optimised by setting the value first and then checking for `not none and (group or logging)` (not 100% sure for the `is not` syntax inside J2 right now)

```suggestion
{%     if aaa_accounting.exec.default.type is arista.avd.defined %}
{%         set aaa_accounting_exec_default_cli = "aaa accounting exec default " ~  aaa_accounting.exec.default.type %}
{%         if aaa_accounting.exec.default.type is not arista.avd.defined('none') and (aaa_accounting.exec.default.group is arista.avd.defined or aaa_accounting.exec.default.logging is arista.avd.defined(true)) %}
{%             if aaa_accounting.exec.default.group is arista.avd.defined %}
{%                 set exec_default_cli = exec_default_cli ~ " group " ~ aaa_accounting.exec.default.group %}
{%             endif %}
{%             if aaa_accounting.exec.default.logging is arista.avd.defined(true) %}
{%                 set exec_default_cli = exec_default_cli ~ " logging" %}
{%             endif %}
{{ exec_default_cli }}
```

Reply:
updated

Reply:
if type is start-stop but did not defined any group or logging then rendering below invalid command by this logic.
aaa accounting commands 3 default start-stop

Reply:
yes thats the poitn of this: `if aaa_accounting.exec.default.type is not arista.avd.defined('none')`

Reply:
ah you mean it should not render  - point taken

Reply:
New suggestion - given that we always need to check for none no matter what first we can do 
1/ check for none - if it is none -> render
2/ check if type is define (it cannot be none) and then check it either group or logging is defined, if so renders, else do nothing

```suggestion
{%     if aaa_accounting.exec.default.type is arista.avd.defined("none") %}
aaa accounting exec default none
{%     elif aaa_accounting.exec.default.type is arista.avd.defined %}
{%         if aaa_accounting.exec.default.group is arista.avd.defined or aaa_accounting.exec.default.logging is arista.avd.defined(true)) %}
{%             set aaa_accounting_exec_default_cli = "aaa accounting exec default " ~  aaa_accounting.exec.default.type %}
{%             if aaa_accounting.exec.default.group is arista.avd.defined %}
{%                 set exec_default_cli = exec_default_cli ~ " group " ~ aaa_accounting.exec.default.group %}
{%             endif %}
{%             if aaa_accounting.exec.default.logging is arista.avd.defined(true) %}
{%                 set exec_default_cli = exec_default_cli ~ " logging" %}
{%             endif %}
{{ exec_default_cli }}
{%         endif %}
{%     endif %}
```